main.go

```diff
@@ -51,6 +55,34 @@ func connectDb() *sql.DB {
 	return db
 }
 
+func migration(upDown string) {
```

Comment:
I get that if the migrations fail we want to stop execution, but its not this function responsibility to stop the service.
Return an error and let the function caller decide what to do with the error.

This is called `error handling`, when a function cannot decide what to do exactly with an error you should give the error more context and return it to the caller.
You can wrap errors using this function:
```go
fmt.Errorf("migrating up: %w", err)
````

https://pkg.go.dev/fmt#Errorf
Wrapping of errors -> https://golangbyexample.com/errorf-function-golang/
https://rollbar.com/blog/golang-wrap-and-unwrap-error/#

Reply:
ok. In this case, I returned the error to the main function, a it then panics with the wrapped error received.